### ISSUE TITLE
Unable to Set Size for md-icon using Iconset and CSS Styling (when using JQuery)

### ISSUE THREAD

**Issue by nkoterba** [angular/material#1679]
I'm trying to use iconsets as described on the angular material demo for [md-icon](https://material.angularjs.org/#/demo/material.components.icon).

I grabbed the demo `core-icons.svg` file.

I included the following code in my angular app:
```javascript
		.config(['$mdIconProvider', function($mdIconProvider) {
			$mdIconProvider
				.iconSet('core', '../styles/images/icons/material-design/core-icons.svg', 24)
		}])
```

My html before styling:
```html
	<md-button ng-click="vm.showNewDialog($event)">
		<md-icon md-svg-src="core:alarm" style="color: yellow"></md-icon>
		<span>New</span>
	</md-button>
```

**Result**:
![image](https://cloud.githubusercontent.com/assets/2508944/6386922/cc92d898-bd4f-11e4-854f-7d62d527cc66.png)

*Now I try to use CSS properties to resize the icon:*

My html:
```html
	<md-button ng-click="vm.showNewDialog($event)">
	         <md-icon md-svg-icon="core:alarm"
		          style="color: yellow; height: 14px; width: 14px"></md-icon>
		<span>New</span>
	</md-button>
```
**Result**:
![image](https://cloud.githubusercontent.com/assets/2508944/6386931/eb53227e-bd4f-11e4-9b16-bd7f222959d2.png)

However, my image does not resize, the md-icon element just gets resized, clipping the md-icon.

Using the chrome inspector, we can see this:
![image](https://cloud.githubusercontent.com/assets/2508944/6386936/fbe4cb7e-bd4f-11e4-993c-d8ad48b4ec09.png)

Oddly, loading icons from svg URLs **does resize the icons** if we apply css styles to them:
```html
	<md-button ng-click="vm.showNewDialog($event)">
		<md-icon md-svg-src="styles/images/icons/material-design/content/ic_add_24px.svg"
				style="color: yellow; width: 10px height: 10px;"></md-icon>
		<span>New</span>
	</md-button>
```
![image](https://cloud.githubusercontent.com/assets/2508944/6386908/847b10de-bd4f-11e4-962c-438f63743615.png)

Has anybody else experienced this or found a solution?

**Comment by ThomasBurleson** [comment]
Close since online demos work with resizing.

**Comment by nkoterba** [comment]
@ThomasBurleson I have discovered the situation where the md-icon resizing **fails**.

At first, I thought it was something strange in my app environment.  It turns out, it simply is the inclusion of jQuery in the webpage (which I'm using instead of jqlite).

I've published a trivial example of how including jQuery in a webpage "breaks" the CSS resizing when using an iconset.  [material-design-broken-icon](https://github.com/nkoterba/material-design-broken-icon).

**Comment by nkoterba** [comment]
@ThomasBurleson Any comment or update on this? Do you plan to re-open this issue?  We've had to revert back to using individual `.svg` images because there's no way to size iconsets when jquery is included in our webapp.

Thanks.

**Comment by dinoboff** [comment]
when jQuery compile the svg elements, it changes their attributes to lower case; `preserveAspectRatio` is set `preserveaspectratio` and `viewBox` is set to `viewbox`.

**Comment by dinoboff** [comment]
Actually, the problem is with jQuery.fn.attr:
http://codepen.io/dinoboff/pen/dPjwBe?editors=101

**Comment by dinoboff** [comment]
@nkoterba You can use  jQuery hook to fix jQuery handling of those attributes:
```
['preserveAspectRatio', 'viewBox'].forEach(function(k) {
  $.attrHooks[k.toLowerCase()] = {
    set: function(el, value) {
      return el.setAttribute(k, value);
    },
    get: function(el) {
      return el.getAttribute(k);
    },
  };
});
```

http://codepen.io/dinoboff/pen/azjXaj

**Comment by ThomasBurleson** [comment]
@dinoboff - You input is very helpful. I think ngMaterial should possbily compensate for these changes also. 
@programmist  - What do you think?

**Comment by programmist** [comment]
@ThomasBurleson If this is indeed a difference between jQuery and jQLite then using the DOM API should fix the issue.  Angular 2.0 won't support jQLite anyway from what I've read.  I'll create a fix and fork @dinoboff's Codepen to test it.  Thanks again for tracking this down @dinoboff.

**Comment by ThomasBurleson** [comment]
@programmist - you rock. Thx.

**Comment by programmist** [comment]
I've got a fix.  I'll submit a PR shortly.

See error condition here:
http://codepen.io/programmist/pen/emLYdj

See issue fixed here:
http://codepen.io/programmist/pen/ogPNzK

**Comment by ThomasBurleson** [comment]
@programmist - is this issue still open ?

**Comment by programmist** [comment]
It's fixed and when I referenced it from my PR it added a closed label (above) that says "This issue will close once pull request #1803 is merged into the 'master' branch."  Maybe that's why it still shows up as open?

**Comment by gtzinos** [comment]
Guys this is a workaround:

.material-icons.md-18 {
font-size: 18px !important;
height: 18px !important;
width: 18px !important;
}
.material-icons.md-24 {
font-size: 24px !important;
height: 24px !important;
width: 24px !important;
}
.material-icons.md-36 {
font-size: 36px !important;
height: 36px !important;
width: 36px !important;
}
.material-icons.md-48 {
font-size: 48px !important;
height: 48px !important;
width: 48px !important;
}
